modules/caddyhttp/templates/tplcontext.go

```diff
@@ -188,6 +188,7 @@ func (c TemplateContext) funcHTTPInclude(uri string) (string, error) {
 		return "", err
 	}
 	virtReq.Host = c.Req.Host
+	virtReq.RemoteAddr = "127.0.0.1"
```

Comment:
Actually, I think it needs to be an `IP:port`, so I suppose we can just set it to some high port and call it a day
```suggestion
	virtReq.RemoteAddr = "127.0.0.1:10000" // https://github.com/caddyserver/caddy/issues/5835
```

Reply:
@francislavoie Could you please explain why we're using a high port (10000) in this context to set `RemoteAddr` to `127.0.0.1:10000`? What specific benefits or considerations does this change bring to our application? Additionally, are there any potential drawbacks or security concerns associated with this approach that I should be aware of?

Reply:
The point of setting `RemoteAddr` here is just so that it's non-empty so that the logic in https://github.com/caddyserver/caddy/blob/4feac4d83cd758c95194090d4f3468373ee342ef/modules/caddyhttp/reverseproxy/reverseproxy.go#L667 does not quit early from having an invalid remote address, and therefore the `X-Forwarded-*` headers will be correctly added to the request sent upstream. The port does not matter at all (because the port on the remote address is always a random port assigned by the kernel networking stack), we never read the port or use it in any way, it just needs to exist for parsing to work properly with `net.SplitHostPort()`.